File: app/src/main/java/wallgram/hd/wallpapers/views/blur/mode/BlurAnimMode.java
package wallgram.hd.wallpapers.views.blur.mode;

import android.widget.ImageView;

public class BlurAnimMode implements AnimMode {

    private static final int DEFAULT_BLUR_RADIUS = 100;
    private int mRadius;

    public BlurAnimMode() {
        this.mRadius = DEFAULT_BLUR_RADIUS;
    }

    public BlurAnimMode(int radius) {
        this.mRadius = radius;
    }

    @Override
    public void transformPage(ImageView ivBg, float position, int direction) {
        float mFraction = (float) Math.cos(2 * Math.PI * position);
        if (mFraction < 0)
            mFraction = 0;
            ivBg.setAlpha(mFraction);
    }

    public int getBlurRadius() {
        return mRadius;
    }
}

File: app/src/main/java/wallgram/hd/wallpapers/views/blur/mode/BlurAnimMode.kt
package wallgram.hd.wallpapers.views.blur.mode

import android.widget.ImageView
import wallgram.hd.wallpapers.views.blur.mode.AnimMode
import wallgram.hd.wallpapers.views.blur.mode.BlurAnimMode
import kotlin.math.cos

class BlurAnimMode(private val blurRadius: Int = DEFAULT_BLUR_RADIUS) : AnimMode {

    override fun transformPage(ivBg: ImageView, position: Float, direction: Int) {
        var mFraction = cos(2 * Math.PI * position).toFloat()
        if (mFraction < 0) mFraction = 0f
        ivBg.alpha = mFraction
    }

    override fun blurRadius() = blurRadius

    companion object {
        private const val DEFAULT_BLUR_RADIUS = 100
    }
}